minimint-api/src/db/mod.rs

```diff
@@ -213,7 +213,7 @@ impl DecodingError {
 #[derive(Debug, Error)]
 pub enum DatabaseError {
     #[error("Underlying Database Error: {0}")]
-    DbError(Box<dyn Error + Send>),
+    DbError(anyhow::Error),
     #[error("Decoding error: {0}")]
     DecodingError(#[from] DecodingError),
 }
```

Comment:
Both cases are Anyhow errors, maybe we can just replace `DatabaseError` with an anyhow error?

Reply:
Yes makes sense ill remove that. sled::Error also impls StdError so we can use anyhow::Error without a problem

Reply:
I think having two variants, both internally being `anyhow::Error` is OK. It does allow the caller to print more informative in formations, or potentially handle different errors differently. So depending on the context here (is decoding error a database error too?) maybe they should be collapsed or not.

Reply:
The `DecodingError` is related to, but not really a database error. In that a `DbError` means there was something wrong with the Database and `DecodingError` could just mean that you supplied the wrong data to create a correct db query. So it makes sense to have it as a`DatabaseError` because if you interact with the db you'll either get that or the decoding succeeds but there is an actual db problem.

Reply:
I think if we have to decide whether these should be  separated by type the question is do we handle them differently...
Right now they are not even distinguished and a `DatabaseError` is always `.expecs("DB error")`.  Since both errors can't be handled by escalating them to the user there is no point in seperating them by type imo. ?